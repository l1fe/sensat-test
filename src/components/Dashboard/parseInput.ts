import { SensorReading } from '../../types';

const parseInput = (str: string): Array<SensorReading | {}> => {
  const parsed: Array<SensorReading | {}> = str
    .split('\n')
    .filter((str) => str.length)
    .map((str) => {
      try {
        const json: SensorReading = JSON.parse(str);
        // Append Z in order to parse the date as UTC
        json.reading_ts = new Date(`${json.reading_ts}Z`);
        return json;
      } catch (err) {
        return {};
      }
    });

  return parsed;
};

export default parseInput;
