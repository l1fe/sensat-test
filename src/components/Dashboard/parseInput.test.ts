import parseInput from './parseInput';

it('should parse input', () => {
  expect(
    parseInput(`
    {"id": "Box-A1-O3", "box_id": "Box-A1", "sensor_type": "O3", "unit": "ppm", "name": "Ozone", "range_l": 0.0, "range_u": 1000.0, "longitude": -0.06507, "latitude": 51.51885, "reading": 672, "reading_ts": "2019-09-10T00:00:00"}\n
    {"id": "Box-A1-NO2", "box_id": "Box-A1", "sensor_type": "NO2", "unit": "ppm", "name": "Nitrogen dioxide", "range_l": 0.0, "range_u": 1000.0, "longitude": -0.06507, "latitude": 51.51885, "reading": 193, "reading_ts": "2019-09-10T00:00:00"}
    `)
  ).toMatchSnapshot();
});
