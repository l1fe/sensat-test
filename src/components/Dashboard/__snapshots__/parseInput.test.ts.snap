// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should parse input 1`] = `
Array [
  Object {
    "box_id": "Box-A1",
    "id": "Box-A1-O3",
    "latitude": 51.51885,
    "longitude": -0.06507,
    "name": "Ozone",
    "range_l": 0,
    "range_u": 1000,
    "reading": 672,
    "reading_ts": "2019-09-10T00:00:00",
    "sensor_type": "O3",
    "unit": "ppm",
  },
  Object {
    "box_id": "Box-A1",
    "id": "Box-A1-NO2",
    "latitude": 51.51885,
    "longitude": -0.06507,
    "name": "Nitrogen dioxide",
    "range_l": 0,
    "range_u": 1000,
    "reading": 193,
    "reading_ts": "2019-09-10T00:00:00",
    "sensor_type": "NO2",
    "unit": "ppm",
  },
  Object {},
]
`;
